src/image/ImageManagerActions.cc

```diff
@@ -562,6 +595,23 @@ int ImageManager::delete_image(int iid, string& error_str)
         {
             NebulaLog::log("ImM",Log::ERROR,
                 "Image could not be removed from DB");
+
+            return -1;
+        }
+
+        int uid = img->get_uid();
+        int gid = img->get_gid();
+        long long size = img->get_size() + img->get_snapshots().total_size();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

We observed that an integer overflow arising for your program. It happens when the outcome of an arithmetic operation surpasses the maximum representable value for the utilized data type. It can lead to unpredictable behavior, security vulnerabilities, and system crashes. To address this vulnerability, it is essential to perform thorough range checking and validation of input values. Employ data types with adequate size to accommodate the anticipated range of values and implement checks to identify and manage potential overflow conditions.